src/Cryptol/TypeCheck/AST.hs

```diff
@@ -112,22 +114,98 @@ groupDecls dg = case dg of
   Recursive ds   -> ds
   NonRecursive d -> [d]
 
+
 data Decl       = Decl { dName        :: !Name
                        , dSignature   :: Schema
                        , dDefinition  :: DeclDef
                        , dPragmas     :: [Pragma]
                        , dInfix       :: !Bool
                        , dFixity      :: Maybe Fixity
                        , dDoc         :: Maybe String
-                       } deriving (Show, Generic, NFData)
+                       } deriving (Generic, NFData, Show)
 
 data DeclDef    = DPrim
                 | DExpr Expr
                   deriving (Show, Generic, NFData)
 
---------------------------------------------------------------------------------
-
+-- ShowAST prints out just enough information to model the program in Coq
```

Comment:
* Perhaps this should go into a separate module
* It might be nice to use the pretty printing combinators, in case we ever want to look at the output.